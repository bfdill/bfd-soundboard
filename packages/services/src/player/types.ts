import { ChildProcess } from 'child_process'

export type DefaultPlayer =
  | 'mplayer'
  | 'afplay'
  | 'mpg123'
  | 'mpg321'
  | 'play'
  | 'omxplayer'
  | 'aplay'
  | 'cmdmp3'
export type What = string | string[]
export type PlayerOptions = {
  player: any
  players?: DefaultPlayer | DefaultPlayer[] | string[]
}
export type PlayerArgs = { [playerName: string]: string[] }
export type PlayOptions = PlayerArgs & { stdio: 'ignore' }
export type Next = Function
export interface IPlayer {
  play: (what: What, options?: PlayOptions | Next, next?: Next) => ChildProcess
}
export type createPlayer = (opts?: PlayerOptions) => IPlayer
