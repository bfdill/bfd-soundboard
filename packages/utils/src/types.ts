export type DeepPartial<T> = {
  [P in keyof T]?: T[P] extends (infer U)[]
    ? DeepPartial<U>[]
    : T[P] extends ReadonlyArray<infer U>
    ? ReadonlyArray<DeepPartial<U>>
    : T[P] extends (string | number | boolean)
    ? T[P]
    : DeepPartial<T[P]>
}
