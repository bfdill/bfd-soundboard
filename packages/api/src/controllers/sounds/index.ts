import * as KoaRouter from 'koa-router'
import { NOT_FOUND, OK } from 'http-status-codes'
import { ISoundService, SoundService } from '@talkboy-9000/sounds'

import { IApplicationContext, ApplicationState } from '../../types'

export interface ISoundsController {
  get: (ctx: IApplicationContext) => void
}

export class SoundsController implements ISoundsController {
  private static instance: ISoundsController | undefined

  constructor(protected readonly soundService: ISoundService) {}

  get = async (context: IApplicationContext) => {
    const logger = context.logger.child({
      controller: {
        name: 'SoundsController',
        method: 'get'
      }
    })
    const sounds = this.soundService.getSounds(logger)

    context.body = { sounds }

    if (sounds === null) {
      logger.warn('no sounds from service')
      context.status = NOT_FOUND
      return
    }

    logger.info({
      soundsLength: sounds.length
    })
    context.status = OK
  }

  static getInstance(): ISoundsController {
    if (this.instance !== undefined) return this.instance

    this.instance = new SoundsController(SoundService.getInstance())

    return this.instance
  }
}

export const getSoundsRouter = () =>
  new KoaRouter<ApplicationState, IApplicationContext>().get(
    '/',
    SoundsController.getInstance().get
  )
