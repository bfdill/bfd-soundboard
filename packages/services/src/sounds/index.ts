import { join } from 'path'

export const PATH_TO_SOUNDS = join(process.cwd(), 'audio')
