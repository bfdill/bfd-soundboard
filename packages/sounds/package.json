{
  "name": "@talkboy-9000/sounds",
  "version": "0.0.1",
  "license": "Unlicense",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "scripts": {
    "build": "tsc -p ./tsconfig.json",
    "clean": "rimraf dist",
    "test": "SUPPRESS_NO_CONFIG_WARNING=false jest --no-cache"
  },
  "devDependencies": {
    "@talkboy-9000/configuration": "^0.0.1",
    "@talkboy-9000/models": "^0.0.1",
    "@talkboy-9000/winston-jest": "^0.0.1"
  }
}
