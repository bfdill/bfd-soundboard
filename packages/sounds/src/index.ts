export { SoundService } from './sounds'
export * from './sounds.types'
