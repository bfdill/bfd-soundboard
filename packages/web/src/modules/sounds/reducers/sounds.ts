import { Reducer } from 'redux'
import { AnyAction, GetSoundsState } from '../../../types'

const reducer: Reducer<GetSoundsState, AnyAction> = (
  state = { sounds: [] },
  action
) => {
  switch (action.type) {
    case 'SUCCESS/getSounds':
      return { sounds: action.payload }
  }

  return state
}

export default reducer
