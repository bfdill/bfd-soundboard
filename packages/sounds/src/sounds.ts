import * as sane from 'sane'
import * as winston from 'winston'
import { sync } from 'glob'
import { join, resolve, sep, basename } from 'path'
import { ISoundService, LOGGER_META } from './sounds.types'
import { Sound } from '@talkboy-9000/models'
import {
  Sounds as SoundConfig,
  ConfigurationService
} from '@talkboy-9000/configuration'

export class SoundService implements ISoundService {
  private static instance: ISoundService | undefined

  protected sounds: Sound[] = []
  protected readonly watcher: sane.Watcher

  constructor(
    public readonly config: SoundConfig,
    protected readonly saneFunction: typeof sane,
    protected readonly logger: winston.Logger
  ) {
    this.addSounds()
    this.watcher = this.createWatch()
  }

  addSounds = () => {
    this.logger.info('add sounds')

    sync(join(this.config.PathToSounds, this.config.FileGlob)).forEach(
      this.addSound
    )
  }

  addSound = (filename: string) => {
    this.logger.debug({ filename })

    this.sounds = [
      ...this.sounds,
      {
        filename,
        id: filename,
        basename: basename(filename)
      }
    ]
      .sort((a, b) => a.filename.localeCompare(b.filename))
      .map((v, i) => {
        v.id = `${i}`
        return v
      })

    this.logger.silly({
      filename,
      sounds: this.sounds
    })
  }

  private createWatch = (): sane.Watcher => {
    return this.saneFunction(this.config.PathToSounds, {
      glob: this.config.FileGlob
    })
      .on('add', (path: string) => {
        this.addSound(path)

        this.logger.debug({
          path,
          message: 'watch:add'
        })
      })
      .on('delete', (path: string) => {
        this.removeSound(path)

        this.logger.debug({
          path,
          message: 'watch:delete'
        })
      })
  }

  getBySoundId = (
    soundId: string,
    parentLogger: winston.Logger
  ): Sound | undefined => {
    const result = this.sounds.find(s => s.id === soundId)
    parentLogger
      .child({
        service: {
          name: 'SoundService',
          method: 'getBySoundId'
        }
      })
      .debug({
        result,
        soundId,
        message: `getBySoundId(${soundId})`
      })
    return result
  }

  getSounds = (parentLogger: winston.Logger): Sound[] => {
    parentLogger
      .child({
        service: {
          name: 'SoundService',
          method: 'getSounds'
        }
      })
      .silly({ sounds: this.sounds })
    return this.sounds
  }

  isPathValid = (filename: string, parentLogger: winston.Logger): boolean => {
    const absCandidate = resolve(filename) + sep
    const result =
      absCandidate.substring(0, this.config.PathToSounds.length) ===
      this.config.PathToSounds

    parentLogger
      .child({
        service: {
          name: 'SoundService',
          method: 'isPathValid'
        }
      })
      .debug({
        filename,
        result,
        message: `isPathValid(${filename})`,
        pathToSounds: this.config.PathToSounds
      })

    return result
  }

  removeSound = (filename: string) => {
    this.logger.debug(`removeSound(${filename})`)

    const soundIndex = this.sounds.findIndex(s => s.filename === filename)

    if (soundIndex === -1) return

    this.sounds = this.sounds
      .slice(0, soundIndex)
      .concat(this.sounds.slice(soundIndex + 1))
      .sort((a, b) => `${a.filename}`.localeCompare(b.filename))
      .map((v, i) => {
        v.id = `${i}`
        return v
      })

    this.logger.silly({
      message: `removeSound(${filename})`,
      sounds: this.sounds
    })
  }

  static getInstance(): ISoundService {
    if (this.instance !== undefined) return this.instance

    this.instance = new SoundService(
      ConfigurationService.getInstance().getSounds(),
      sane,
      winston.createLogger({ defaultMeta: LOGGER_META })
    )

    return this.instance
  }
}
