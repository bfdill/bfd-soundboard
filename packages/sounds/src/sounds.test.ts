import * as sane from 'sane'
import { join } from 'path'
import { getJestLogger, IJestLogger } from '@talkboy-9000/winston-jest'
import { SoundService } from './sounds'
import { ISoundService } from './sounds.types'

describe('modules -> sounds', () => {
  const PATH_TO_SOUNDS = '/tmp/fakeaf'
  const saneOnMock = jest.fn()
  const saneWatcherMock = {
    on: saneOnMock
  }
  const saneFunctionMock = (_: string, _1?: sane.Options): sane.Watcher => {
    return saneWatcherMock as any
  }
  const jestLogger: IJestLogger = getJestLogger()
  let soundService: ISoundService
  let baseSoundService: SoundService

  beforeEach(() => {
    saneOnMock.mockReturnValue(saneWatcherMock)
    soundService = new SoundService(
      PATH_TO_SOUNDS,
      saneFunctionMock as any,
      jestLogger.logger
    )
    baseSoundService = soundService as SoundService
  })

  describe('addSound', () => {
    test('id = ordinal position by filename alpha', () => {
      expect(soundService.getSounds(jestLogger.logger)).toHaveLength(0)
      baseSoundService.addSound('/tmp/fake_sound_0.mp3')

      expect(soundService.getSounds(jestLogger.logger)).toHaveLength(1)
      expect(soundService.getSounds(jestLogger.logger)).toMatchSnapshot()

      baseSoundService.addSound('/tmp/a_fake_sound_0.mp3')

      expect(soundService.getSounds(jestLogger.logger)).toHaveLength(2)
      expect(soundService.getSounds(jestLogger.logger)).toMatchSnapshot()

      jestLogger.callsMatchSnapshot()
    })
  })

  describe('createWatch', () => {
    test('wires add and delete on instantiation', () => {
      saneOnMock.mock.calls.forEach(call => {
        expect(call).toMatchSnapshot()
      })

      jestLogger.callsMatchSnapshot()
    })

    test('add, adds', () => {
      const addSoundSpy = jest.spyOn(baseSoundService, 'addSound')
      const fn = saneOnMock.mock.calls[0][1]
      fn('add test')
      expect(addSoundSpy).toHaveBeenCalledWith('add test')
      jestLogger.callsMatchSnapshot()
    })

    test('delete, removes', () => {
      const removeSoundSpy = jest.spyOn(baseSoundService, 'removeSound')
      const fn = saneOnMock.mock.calls[1][1]
      fn('remove test')
      expect(removeSoundSpy).toHaveBeenCalledWith('remove test')
      jestLogger.callsMatchSnapshot()
    })
  })

  describe('getBySoundId', () => {
    test.each<any>([undefined, null, '', '0', 'dog'])(
      "getBySoundId('%s')",
      (soundId: string) => {
        baseSoundService.addSound('/tmp/fake_sound_0.mp3')
        jestLogger.transport.mock.mockReset()

        expect(
          soundService.getBySoundId(soundId, jestLogger.logger)
        ).toMatchSnapshot()
        jestLogger.callsMatchSnapshot()
      }
    )
  })

  describe('getSounds', () => {
    test('no sounds', () => {
      expect(soundService.getSounds(jestLogger.logger)).toHaveLength(0)

      jestLogger.callsMatchSnapshot()
    })

    test('two sounds', () => {
      expect(soundService.getSounds(jestLogger.logger)).toHaveLength(0)

      baseSoundService.addSound('/tmp/fake_sound_0.mp3')
      baseSoundService.addSound('/tmp/a_fake_sound.mp3')
      jestLogger.transport.mock.mockReset()

      const sounds = soundService.getSounds(jestLogger.logger)
      expect(sounds).toHaveLength(2)
      expect(sounds).toMatchSnapshot()

      jestLogger.callsMatchSnapshot()
    })
  })

  describe('isPathValid', () => {
    test.each<[string, boolean]>([
      ['/tmp/invalid_af', false],
      [join(PATH_TO_SOUNDS, 'too_legit'), true]
    ])('validity test', (filename: string, expected) => {
      const actual = soundService.isPathValid(filename, jestLogger.logger)

      expect(actual).toEqual(expected)
    })
  })

  describe('removeSound', () => {
    beforeEach(() => {
      baseSoundService.addSound('a')
      baseSoundService.addSound('b')
      baseSoundService.addSound('c')
      jestLogger.transport.mock.mockReset()
    })

    test('unknown file, no action', () => {
      const expected = baseSoundService.getSounds(jestLogger.logger)
      jestLogger.transport.mock.mockReset()

      baseSoundService.removeSound('f')
      jestLogger.callsMatchSnapshot()

      const actual = baseSoundService.getSounds(jestLogger.logger)
      expect(actual).toEqual(expected)
      expect(actual).toHaveLength(3)
    })

    test.each(['a', 'b', 'c'])(
      'removes file and re-sorts remaining items',
      (filename: string) => {
        baseSoundService.removeSound(filename)
        jestLogger.callsMatchSnapshot()

        expect(baseSoundService.getSounds(jestLogger.logger)).toMatchSnapshot()
      }
    )
  })
})
