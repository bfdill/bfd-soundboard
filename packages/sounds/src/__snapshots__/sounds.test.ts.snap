// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules -> sounds addSound id = ordinal position by filename alpha 1`] = `
Array [
  Object {
    "basename": "fake_sound_0.mp3",
    "filename": "/tmp/fake_sound_0.mp3",
    "id": "0",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 2`] = `
Array [
  Object {
    "basename": "a_fake_sound_0.mp3",
    "filename": "/tmp/a_fake_sound_0.mp3",
    "id": "0",
  },
  Object {
    "basename": "fake_sound_0.mp3",
    "filename": "/tmp/fake_sound_0.mp3",
    "id": "1",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 3`] = `
Array [
  Object {
    "level": "info",
    "message": "add sounds",
    Symbol(level): "info",
    Symbol(message): "{\\"message\\":\\"add sounds\\",\\"level\\":\\"info\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 4`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "sounds": Array [],
    },
    "service": Object {
      "method": "getSounds",
      "name": "SoundService",
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getSounds\\"},\\"message\\":{\\"sounds\\":[]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 5`] = `
Array [
  Object {
    "level": "debug",
    "message": Object {
      "filename": "/tmp/fake_sound_0.mp3",
    },
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\"},\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 6`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "filename": "/tmp/fake_sound_0.mp3",
      "sounds": Array [
        Object {
          "basename": "fake_sound_0.mp3",
          "filename": "/tmp/fake_sound_0.mp3",
          "id": "1",
        },
      ],
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"sounds\\":[{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"fake_sound_0.mp3\\"}]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 7`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "sounds": Array [
        Object {
          "basename": "fake_sound_0.mp3",
          "filename": "/tmp/fake_sound_0.mp3",
          "id": "1",
        },
      ],
    },
    "service": Object {
      "method": "getSounds",
      "name": "SoundService",
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getSounds\\"},\\"message\\":{\\"sounds\\":[{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"fake_sound_0.mp3\\"}]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 8`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "sounds": Array [
        Object {
          "basename": "fake_sound_0.mp3",
          "filename": "/tmp/fake_sound_0.mp3",
          "id": "1",
        },
      ],
    },
    "service": Object {
      "method": "getSounds",
      "name": "SoundService",
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getSounds\\"},\\"message\\":{\\"sounds\\":[{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"fake_sound_0.mp3\\"}]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 9`] = `
Array [
  Object {
    "level": "debug",
    "message": Object {
      "filename": "/tmp/a_fake_sound_0.mp3",
    },
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":{\\"filename\\":\\"/tmp/a_fake_sound_0.mp3\\"},\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 10`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "filename": "/tmp/a_fake_sound_0.mp3",
      "sounds": Array [
        Object {
          "basename": "a_fake_sound_0.mp3",
          "filename": "/tmp/a_fake_sound_0.mp3",
          "id": "0",
        },
        Object {
          "basename": "fake_sound_0.mp3",
          "filename": "/tmp/fake_sound_0.mp3",
          "id": "1",
        },
      ],
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":{\\"filename\\":\\"/tmp/a_fake_sound_0.mp3\\",\\"sounds\\":[{\\"filename\\":\\"/tmp/a_fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"a_fake_sound_0.mp3\\"},{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"1\\",\\"basename\\":\\"fake_sound_0.mp3\\"}]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 11`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "sounds": Array [
        Object {
          "basename": "a_fake_sound_0.mp3",
          "filename": "/tmp/a_fake_sound_0.mp3",
          "id": "0",
        },
        Object {
          "basename": "fake_sound_0.mp3",
          "filename": "/tmp/fake_sound_0.mp3",
          "id": "1",
        },
      ],
    },
    "service": Object {
      "method": "getSounds",
      "name": "SoundService",
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getSounds\\"},\\"message\\":{\\"sounds\\":[{\\"filename\\":\\"/tmp/a_fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"a_fake_sound_0.mp3\\"},{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"1\\",\\"basename\\":\\"fake_sound_0.mp3\\"}]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 12`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "sounds": Array [
        Object {
          "basename": "a_fake_sound_0.mp3",
          "filename": "/tmp/a_fake_sound_0.mp3",
          "id": "0",
        },
        Object {
          "basename": "fake_sound_0.mp3",
          "filename": "/tmp/fake_sound_0.mp3",
          "id": "1",
        },
      ],
    },
    "service": Object {
      "method": "getSounds",
      "name": "SoundService",
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getSounds\\"},\\"message\\":{\\"sounds\\":[{\\"filename\\":\\"/tmp/a_fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"a_fake_sound_0.mp3\\"},{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"1\\",\\"basename\\":\\"fake_sound_0.mp3\\"}]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds createWatch add, adds 1`] = `
Array [
  Object {
    "level": "info",
    "message": "add sounds",
    Symbol(level): "info",
    Symbol(message): "{\\"message\\":\\"add sounds\\",\\"level\\":\\"info\\"}",
  },
]
`;

exports[`modules -> sounds createWatch add, adds 2`] = `
Array [
  Object {
    "level": "debug",
    "message": Object {
      "filename": "add test",
    },
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":{\\"filename\\":\\"add test\\"},\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds createWatch add, adds 3`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "filename": "add test",
      "sounds": Array [
        Object {
          "basename": "add test",
          "filename": "add test",
          "id": "0",
        },
      ],
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":{\\"filename\\":\\"add test\\",\\"sounds\\":[{\\"filename\\":\\"add test\\",\\"id\\":\\"0\\",\\"basename\\":\\"add test\\"}]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds createWatch add, adds 4`] = `
Array [
  Object {
    "level": "debug",
    "message": "watch:add",
    "path": "add test",
    Symbol(level): "debug",
    Symbol(message): "{\\"path\\":\\"add test\\",\\"message\\":\\"watch:add\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds createWatch delete, removes 1`] = `
Array [
  Object {
    "level": "info",
    "message": "add sounds",
    Symbol(level): "info",
    Symbol(message): "{\\"message\\":\\"add sounds\\",\\"level\\":\\"info\\"}",
  },
]
`;

exports[`modules -> sounds createWatch delete, removes 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(remove test)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(remove test)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds createWatch delete, removes 3`] = `
Array [
  Object {
    "level": "debug",
    "message": "watch:delete",
    "path": "remove test",
    Symbol(level): "debug",
    Symbol(message): "{\\"path\\":\\"remove test\\",\\"message\\":\\"watch:delete\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds createWatch wires add and delete on instantiation 1`] = `
Array [
  "add",
  [Function],
]
`;

exports[`modules -> sounds createWatch wires add and delete on instantiation 2`] = `
Array [
  "delete",
  [Function],
]
`;

exports[`modules -> sounds createWatch wires add and delete on instantiation 3`] = `
Array [
  Object {
    "level": "info",
    "message": "add sounds",
    Symbol(level): "info",
    Symbol(message): "{\\"message\\":\\"add sounds\\",\\"level\\":\\"info\\"}",
  },
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('') 1`] = `undefined`;

exports[`modules -> sounds getBySoundId getBySoundId('') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId()",
    "result": undefined,
    "service": Object {
      "method": "getBySoundId",
      "name": "SoundService",
    },
    "soundId": "",
    Symbol(level): "debug",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getBySoundId\\"},\\"soundId\\":\\"\\",\\"message\\":\\"getBySoundId()\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('0') 1`] = `
Object {
  "basename": "fake_sound_0.mp3",
  "filename": "/tmp/fake_sound_0.mp3",
  "id": "0",
}
`;

exports[`modules -> sounds getBySoundId getBySoundId('0') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId(0)",
    "result": Object {
      "basename": "fake_sound_0.mp3",
      "filename": "/tmp/fake_sound_0.mp3",
      "id": "0",
    },
    "service": Object {
      "method": "getBySoundId",
      "name": "SoundService",
    },
    "soundId": "0",
    Symbol(level): "debug",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getBySoundId\\"},\\"result\\":{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"fake_sound_0.mp3\\"},\\"soundId\\":\\"0\\",\\"message\\":\\"getBySoundId(0)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('dog') 1`] = `undefined`;

exports[`modules -> sounds getBySoundId getBySoundId('dog') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId(dog)",
    "result": undefined,
    "service": Object {
      "method": "getBySoundId",
      "name": "SoundService",
    },
    "soundId": "dog",
    Symbol(level): "debug",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getBySoundId\\"},\\"soundId\\":\\"dog\\",\\"message\\":\\"getBySoundId(dog)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('null') 1`] = `undefined`;

exports[`modules -> sounds getBySoundId getBySoundId('null') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId(null)",
    "result": undefined,
    "service": Object {
      "method": "getBySoundId",
      "name": "SoundService",
    },
    "soundId": null,
    Symbol(level): "debug",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getBySoundId\\"},\\"soundId\\":null,\\"message\\":\\"getBySoundId(null)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('undefined') 1`] = `undefined`;

exports[`modules -> sounds getBySoundId getBySoundId('undefined') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId(undefined)",
    "result": undefined,
    "service": Object {
      "method": "getBySoundId",
      "name": "SoundService",
    },
    "soundId": undefined,
    Symbol(level): "debug",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getBySoundId\\"},\\"message\\":\\"getBySoundId(undefined)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getSounds no sounds 1`] = `
Array [
  Object {
    "level": "info",
    "message": "add sounds",
    Symbol(level): "info",
    Symbol(message): "{\\"message\\":\\"add sounds\\",\\"level\\":\\"info\\"}",
  },
]
`;

exports[`modules -> sounds getSounds no sounds 2`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "sounds": Array [],
    },
    "service": Object {
      "method": "getSounds",
      "name": "SoundService",
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getSounds\\"},\\"message\\":{\\"sounds\\":[]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds getSounds two sounds 1`] = `
Array [
  Object {
    "basename": "a_fake_sound.mp3",
    "filename": "/tmp/a_fake_sound.mp3",
    "id": "0",
  },
  Object {
    "basename": "fake_sound_0.mp3",
    "filename": "/tmp/fake_sound_0.mp3",
    "id": "1",
  },
]
`;

exports[`modules -> sounds getSounds two sounds 2`] = `
Array [
  Object {
    "level": "silly",
    "message": Object {
      "sounds": Array [
        Object {
          "basename": "a_fake_sound.mp3",
          "filename": "/tmp/a_fake_sound.mp3",
          "id": "0",
        },
        Object {
          "basename": "fake_sound_0.mp3",
          "filename": "/tmp/fake_sound_0.mp3",
          "id": "1",
        },
      ],
    },
    "service": Object {
      "method": "getSounds",
      "name": "SoundService",
    },
    Symbol(level): "silly",
    Symbol(message): "{\\"service\\":{\\"name\\":\\"SoundService\\",\\"method\\":\\"getSounds\\"},\\"message\\":{\\"sounds\\":[{\\"filename\\":\\"/tmp/a_fake_sound.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"a_fake_sound.mp3\\"},{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"1\\",\\"basename\\":\\"fake_sound_0.mp3\\"}]},\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 1`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(a)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(a)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 2`] = `
Array [
  Object {
    "level": "silly",
    "message": "removeSound(a)",
    "sounds": Array [
      Object {
        "basename": "b",
        "filename": "b",
        "id": "0",
      },
      Object {
        "basename": "c",
        "filename": "c",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"removeSound(a)\\",\\"sounds\\":[{\\"filename\\":\\"b\\",\\"id\\":\\"0\\",\\"basename\\":\\"b\\"},{\\"filename\\":\\"c\\",\\"id\\":\\"1\\",\\"basename\\":\\"c\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 3`] = `
Array [
  Object {
    "basename": "b",
    "filename": "b",
    "id": "0",
  },
  Object {
    "basename": "c",
    "filename": "c",
    "id": "1",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 4`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(b)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(b)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 5`] = `
Array [
  Object {
    "level": "silly",
    "message": "removeSound(b)",
    "sounds": Array [
      Object {
        "basename": "a",
        "filename": "a",
        "id": "0",
      },
      Object {
        "basename": "c",
        "filename": "c",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"removeSound(b)\\",\\"sounds\\":[{\\"filename\\":\\"a\\",\\"id\\":\\"0\\",\\"basename\\":\\"a\\"},{\\"filename\\":\\"c\\",\\"id\\":\\"1\\",\\"basename\\":\\"c\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 6`] = `
Array [
  Object {
    "basename": "a",
    "filename": "a",
    "id": "0",
  },
  Object {
    "basename": "c",
    "filename": "c",
    "id": "1",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 7`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(c)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(c)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 8`] = `
Array [
  Object {
    "level": "silly",
    "message": "removeSound(c)",
    "sounds": Array [
      Object {
        "basename": "a",
        "filename": "a",
        "id": "0",
      },
      Object {
        "basename": "b",
        "filename": "b",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"removeSound(c)\\",\\"sounds\\":[{\\"filename\\":\\"a\\",\\"id\\":\\"0\\",\\"basename\\":\\"a\\"},{\\"filename\\":\\"b\\",\\"id\\":\\"1\\",\\"basename\\":\\"b\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 9`] = `
Array [
  Object {
    "basename": "a",
    "filename": "a",
    "id": "0",
  },
  Object {
    "basename": "b",
    "filename": "b",
    "id": "1",
  },
]
`;

exports[`modules -> sounds removeSound unknown file, no action 1`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(f)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(f)\\",\\"level\\":\\"debug\\"}",
  },
]
`;
